portfolio/src/main/java/com/google/sps/servlets/AnimeRecServlet.java

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.AnimeRec;
+import com.google.gson.Gson;
+import java.util.*;
+import java.io.IOException;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns anime recommendation content.*/
+@WebServlet("/toptenrecommendationdata")
+public class AnimeRecServlet extends HttpServlet {
+  ArrayList<AnimeRec> recommendationsList = new ArrayList<>();
+  String convertRecommendationListToJson() {
+    Gson gson = new Gson();
+    return gson.toJson(recommendationsList);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
```

Comment:
When debugging, its helpful to print out statements to verify the current state and whether methods are being called. So for your server, you can print `recommendationsList` to check that items are added to it.

You can use `System.out.println("doGet: " + recommendationsList);` and `System.out.println("doPost: " + recommendationsList);` in the corresponding methods, then check the console for the output.

Reply:
Noted, I will try that and decide what to do from there. Thanks!